### DISCUSSION TITLE
Can Highs be compiled to ARM architecture version?

### DISCUSSION THREAD

**Discussion by HeisenbergPPT** [ERGO-Code/HiGHS#1550] [General]
Can Highs be compiled to ARM architecture version?
I have tried to use a cross-compilation, but doesn't work.

**Comment by jajhall** [comment]
I don't know, but I guess @galabovaa does

**Comment by galabovaa** [comment]
Are you using CMake to compile the C++ code? Which version? Which operating system are you using, Windows / Linux / MacOS? 

The python build is currently WIP but C++ should be building fine on ARM architectures.

**Reply by HeisenbergPPT** [reply]
Yes, I am compile C++ code, gcc 9.4.0, on Linux.

**Reply by HeisenbergPPT** [reply]
I got it. Just need to add -fexception and -pthread in the cross-compiling toolchain file, then it works.

**Comment by odow** [comment]
> Can Highs be compiled to ARM architecture version?

Yes. You can find a variety of pre-build executables here: https://github.com/JuliaBinaryWrappers/HiGHSstatic_jll.jl/releases.

See, for example, [HiGHSstatic.v1.6.0.armv7l-linux-gnueabihf-cxx11.tar.gz](https://github.com/JuliaBinaryWrappers/HiGHSstatic_jll.jl/releases/download/HiGHSstatic-v1.6.0%2B0/HiGHSstatic.v1.6.0.armv7l-linux-gnueabihf-cxx11.tar.gz) and [HiGHSstatic.v1.6.0.aarch64-linux-gnu-cxx11.tar.gz](https://github.com/JuliaBinaryWrappers/HiGHSstatic_jll.jl/releases/download/HiGHSstatic-v1.6.0%2B0/HiGHSstatic.v1.6.0.aarch64-linux-gnu-cxx11.tar.gz)

**Reply by HeisenbergPPT** [reply]
That look promising! let me try.

**Comment by HeisenbergPPT** [comment]
Just need to add -fexception and -pthread in the cross-compiling toolchain file,
and then:
cmake -DCMAKE_TOOLCHAIN_FILE=xxxxx.cmake(toolchain file for ARM version) ..
make -j32

it works.
